app/models/user.rb

```diff
@@ -1,8 +1,35 @@
+require "net/http"
+require "uri"
+
 class User < ActiveRecord::Base
   has_secure_password
 
+  enum authy_status: [:unverified, :onetouch, :sms, :token, :approved, :denied]
   validates :email,  presence: true, format: { with: /\A.+@.+$\Z/ }, uniqueness: true
   validates :name, presence: true
   validates :country_code, presence: true
   validates :phone_number, presence: true
+
+  def send_one_touch
+    response = Authy::OneTouch.send_approval_request(
+      id: self.authy_id,
+      message: "Request to Login to Twilio demo app",
+      details: {
+        'Email Address' => self.email,
+        'Amount' => '10 BTC',
+      }
+    )
+    set_status_and_uid(response.body)
+
+    return response.body
+  end
+
+  private
+
+  def set_status_and_uid(response)
+    response = JSON.parse(response)
```

Comment:
If you're using the library now, this doesn't need to happen. You should be able to call `response.approval_request` on the original `response` object. I still feel though this should be in the sessions controller, like this:

```ruby
  def create
    @user = User.find_by(email: params[:email])
    if @user && @user.authenticate(params[:password])
      session[:pre_2fa_auth_user_id] = @user.id
      
      # Try to verify with OneTouch
      one_touch = Authy::OneTouch.send_approval_request(
        id: @user.authy_id,
        message: "Request to Login to Twilio demo app",
        details: {
          'Email Address' => @user.email,
          'Amount' => '10 BTC',
        }
      )
      status = response.approval_request ? :onetouch : :sms
      @user.update(authy_status: status)

      # Respond to the ajax call that requested this with the approval request body
      render json: one_touch.body
    else
      @user ||= User.new(email: params[:email])
      render :new
    end
  end
```